### PR TITLE
Add probe periodically performing DNS lookups to probes binary

### PR DESCRIPTION
**Pull request by oxddr**
ref #612

/assign @wojtek-t @mm4tt

### CODE REVIEW

**Comment by mm4tt** [comment]
./verify/verify-flags-underscore.py fails, please fix

File: probes/pkg/dns/probe.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	})
+
+	probe dnsProbe
+)
+
+func init() {
+	prometheus.MustRegister(inClusterDNSLatency, inClusterDNSLookupCount, inClusterDNSLookupError)
+
+	flag.StringVar(&probe.lookupURL, "dns-probe-url", "", "Name of a Service to lookup")
+	flag.DurationVar(&probe.probingInterval, "probing-interval", 1*time.Second, "Interval between DNS lookups")
```
Comment (mm4tt) [inline probes/pkg/dns/probe.go]: dns-probing-interval?

File: probes/pkg/dns/probe.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+// Run periodically does DNS lookups.
+// Lookup's interval and URL are configurable via flags.
+func Run() {
+	if probe.lookupURL == "" {
+		klog.Fatal("--dns-probe-url has not been set")
+	}
+	probe.run()
+}
+
+func (p *dnsProbe) run() {
```
Comment (wojtek-t) [inline probes/pkg/dns/probe.go]: You may want to pass stopCh to it.

Comment (oxddr) [reply probes/pkg/dns/probe.go]: This is not used by other probes. I'd rather add it in another PR.

File: probes/pkg/dns/probe.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+		Help:      "Counter of DNS lookups made by dns-prober.",
+	})
+	// inClusterDNSLookupError counts failed DNS lookups done by prober.
+	inClusterDNSLookupError = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: namespace,
+		Name:      "in_cluster_dns_lookup_error",
+		Help:      "Counter of DNS lookups made by dns-prober that failed",
+	})
+
+	probe dnsProbe
```
[the remark this answers is not in the document]
Comment (oxddr) [reply probes/pkg/dns/probe.go]: Switched back to pure flags.

**Comment by oxddr** [comment]
Ping

File: probes/pkg/dns/probe.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+		Help:      "Counter of DNS lookups made by dns-prober that failed",
+	})
+
+	url      = flag.String("dns-probe-url", "", "Name of a Service to lookup")
+	interval = flag.Duration("dns-probe-interval", 1*time.Second, "Interval between DNS lookups")
+)
+
+func init() {
+	prometheus.MustRegister(inClusterDNSLatency, inClusterDNSLookupCount, inClusterDNSLookupError)
+
```
Comment (wojtek-t) [inline probes/pkg/dns/probe.go]: nit: remove empty line

File: probes/pkg/dns/probe.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+	"flag"
+	"net"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/klog"
+)
+
+const (
+	namespace = "probes"
```
Comment (wojtek-t) [inline probes/pkg/dns/probe.go]: This should probably be some constant at the level of probes/ directory.

**Comment by oxddr** [comment]
If there is no outstanding issues with this PR, could you please approve it?

**Pull request merged**